kubemarine/core/defaults.py

```diff
@@ -93,6 +93,10 @@ def apply_defaults(inventory, cluster):
             raise Exception('Node name \"%s\" contains invalid characters. A DNS-1123 subdomain must consist of lower '
                             'case alphanumeric characters, \'-\' or \'.\'' % node_name)
 
+        # Use internal address if address is not defined
+        if node.get('address') is None and node.get("internal_address") is not None:
+            cluster.inventory["nodes"][i]["address"] = node.get("internal_address")
```

Comment:
Presence of address property influences the behaviour of Kubemarine. For example, see defaults.append_controlplain and system.generate_etc_hosts_config. Won't it be safe if we just check presence of the property where necessary?
Also I suggest to develop unit test which ensure that enrichment is successful without address property.
If you still want to assign address = internal_address and have explanation why it is safe, this makes many places in code doing unnecessary checks (for example, the same places in defaults and system mentioned).

Reply:
Yes, I see. I decided not to fill this value from internal address during enrichment. 
I've checked all places, there 'address' appears and found two places, where we don't have such check. Those places were fixed.

Reply:
Thank you, and what about this
> Also I suggest to develop unit test which ensure that enrichment is successful without address property.
Based on the implementation, I conclude that tests of generation of final and finalized inventory are also desirable.

Reply:
Test was added